### ISSUE TITLE
Cascades

### ISSUE THREAD

**Issue by simlei** [jcryptool/core#207]
I got a report about the algorithm cascades in JCT

https://github.com/jcryptool/crypto/issues/315

The question is, whether we want to maintain this feature (Action cascades) as-is.
The main problem I see is, that this feature is placed so prominently when JCT starts, but was never perceived by the developers as a central feature. The result is, that it does not rhyme too well in execution with a lot of JCT functionality. 

This is how the default perspective looks like with  `Documents/.jcryptool` deleted previously:

![image](https://user-images.githubusercontent.com/982379/83769381-0ea5e800-a680-11ea-85a6-650f5dee05b6.png)

IMHO, no one needs to see the action cascade recording tool right away and I think we could hide it away if not disable it. 

Furthermore I see architectural problems with it going forward from v 1.0:
- every "actionable" thing the user does in JCT based on editor text must be recordable and repeatable by this system. 
- If not, a recorded action chain will be potentially worthless for playback because of non-recorded parts -> inconsistent behavior. 
- This makes it a feature that is only "stable" if potentially every part of JCT actively chips in with carrying this stability. 
- That makes it a potential source of regression bugs that need fixing if the action cascades are considered a quality goal of JCT.

I think it's too much of a burden for the little actual value it provides for users (I have yet to hear of JCT power users that need action cascades. Our modularity and granularity of actions is just not there yet)

Therefore, I lean towards removing the feature altogether from the release of v 1.0. 
I'm open to opinions about compromises, e.g. making it a much less prominent feature, where the problem of technical debt incurred vs. value to the user would be addressed.

**Comment by simlei** [comment]
This would also have to be addressed in the JCrypTool user guide, for the record

**Comment by grthor** [comment]
Try to fix the bug til version 1.0.

If we do not get it done, remove this feature for version 1.0, create an issue and fix it later.

**Comment by simlei** [comment]
Recorded cascades seem to ignore some portion of the parameters. When recording, the en-/decryption of the example file with Caesar went fine, so that the decrypted text was identical to the original; when played back, the decrypted text is filtered of whitespace and line breaks (otherwise the same).

![cascades1](https://user-images.githubusercontent.com/982379/86124777-ee7d1380-badb-11ea-89ab-127dc17cd70c.png)
![cascades3](https://user-images.githubusercontent.com/982379/86124782-f0df6d80-badb-11ea-8f97-981c42f02d0e.png)

**Comment by simlei** [comment]
> Try to fix the bug til version 1.0.
> If we do not get it done, remove this feature for version 1.0, create an issue and fix it later.

As for the bug, jcryptool/crypto#315, there is no error anymore, so I closed it. however, I had to observe the above problem which is not ideal and probably needs testing+fixing as well. Two points of interest to look into:
 - Are text transformations (Classical algorithms -> page 2 of the Wizard) played back correctly?
 - What is the root of the aforementioned problem, if not text transformations, where decrypted Caesar and original text are not identical?
